### ISSUE TITLE
Storagerouter offline results in backend not loadable

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#679]
**INFO:** The storage router that is offline contains a (part of a) local flash backend.

When looking at the GUI I see this: https://s32.postimg.org/huv0gd639/volumedrivernode_offline_backend_offline.png

When trying to load the backends, I see this in `ovs-webapp-api` log:

```
2016-06-29 10:52:11 36400 +0200 - ctrl-roub-01 - 27740/140621642465360 - log/api - 20476 - INFO - [albabackends.retrieve] - 730bf425-8cf9-4a96-97c4-c0c776c71d06 - [] - {"pk": "d5c86718-b016-4cf0-ae2f-c272a41bd5d4"} - {"cookies": {}, "meta": {"HTTP_AUTHORIZATION": "Bearer 9*~2g{o\"kA}pP>blqee/]|YU7qsK.RU~*W[k|2s,O,sU=|mkvJUw_+7q7flgi0N0", "wsgi.multiprocess": "True", "HTTP_REFERER": "https://51.254.197.51/", "SERVER_PROTOCOL": "HTTP/1.0", "SERVER_SOFTWARE": "gunicorn/17.5", "SCRIPT_NAME": "/api", "REQUEST_METHOD": "GET", "PATH_INFO": "/alba/backends/d5c86718-b016-4cf0-ae2f-c272a41bd5d4/", "HTTP_X_FORWARDED_SSL": "on", "QUERY_STRING": "contents=_dynamics%2C_relations&timestamp=1467190331281", "HTTP_X_REAL_IP": "212.88.230.96", "HTTP_USER_AGENT": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.110 Safari/537.36", "HTTP_CONNECTION": "close", "SERVER_NAME": "51.254.197.51", "REMOTE_PORT": "37834", "wsgi.url_scheme": "https", "SERVER_PORT": "443", "HTTP_X_SCHEME": "https", "HTTP_X_REQUESTED_WITH": "XMLHttpRequest", "wsgi.input": "<gunicorn.http.body.Body object at 0x7fe5070e3b90>", "HTTP_DNT": "1", "HTTP_HOST": "51.254.197.51", "wsgi.multithread": "False", "HTTP_ACCEPT": "application/json; version=*", "wsgi.version": "(1, 0)", "RAW_URI": "/alba/backends/d5c86718-b016-4cf0-ae2f-c272a41bd5d4/?contents=_dynamics%2C_relations&timestamp=1467190331281", "wsgi.run_once": "False", "wsgi.errors": "<open file '<stderr>', mode 'w' at 0x7fe51d47b1e0>", "REMOTE_ADDR": "127.0.0.1", "HTTP_ACCEPT_LANGUAGE": "nl,en-US;q=0.8,en;q=0.6", "gunicorn.socket": "<socket fileno=38 sock=127.0.0.1:8002 peer=127.0.0.1:37834>", "CONTENT_TYPE": "application/json", "wsgi.file_wrapper": "gunicorn.http.wsgi.FileWrapper", "CSRF_COOKIE": "98UIJfycjB5C3NGwVXnJfmhFcG7NE7Rb", "HTTP_ACCEPT_ENCODING": "gzip, deflate, sdch"}, "request": {"timestamp": "1467190331281", "contents": "_dynamics,_relations"}}
2016-06-29 10:52:11 56700 +0200 - ctrl-roub-01 - 27740/140621642465360 - api/middleware - 20477 - ERROR - An unhandled exception occurred:  
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 78, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 399, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 396, in dispatch
    response = handler(request, *args, **kwargs)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 427, in new_function
    return_value = f(*args, **kwargs)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 73, in new_function
    return f(*args, **kw)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 290, in new_function
    return Response(FullSerializer(object_type, contents=contents, instance=obj).data, status=status.HTTP_200_OK)
  File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 573, in data
    self._data = self.to_native(obj)
  File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 349, in to_native
    value = field.field_to_native(obj, field_name)
  File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 198, in field_to_native
    value = get_component(value, component)
  File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 56, in get_component
    val = getattr(obj, attr_name)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 339, in <lambda>
    fget = lambda s: s._get_dynamic_property(dynamic)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 408, in _get_dynamic_property
    return self._backend_property(data_loader, dynamic)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 898, in _backend_property
    dynamic_data = function()
  File "/opt/OpenvStorage/ovs/dal/hybrids/albabackend.py", line 262, in _ns_statistics
    vdisk_dataset[vpool] = vpool.storagedriver_client.list_volumes()
ClusterNotReachableException:  
```

```
2016-06-29 10:52:25 06100 +0200 - ctrl-roub-01 - 27729/140621633306128 - log/api - 9555 - INFO - [backends.retrieve] - 730bf425-8cf9-4a96-97c4-c0c776c71d06 
- [] - {"pk": "0a0609af-7ca6-4254-8677-66d4b1b0ef56"} - {"cookies": {}, "meta": {"HTTP_AUTHORIZATION": "Bearer 9*~2g{o\"kA}pP>blqee/]|YU7qsK.RU~*W[k|2s,O,sU
=|mkvJUw_+7q7flgi0N0", "wsgi.multiprocess": "True", "HTTP_REFERER": "https://51.254.197.51/", "SERVER_PROTOCOL": "HTTP/1.0", "SERVER_SOFTWARE": "gunicorn/17
.5", "SCRIPT_NAME": "/api", "REQUEST_METHOD": "GET", "PATH_INFO": "/backends/0a0609af-7ca6-4254-8677-66d4b1b0ef56/", "HTTP_X_FORWARDED_SSL": "on", "QUERY_ST
RING": "contents=_relations%2Cregular_domains&timestamp=1467190344677", "HTTP_X_REAL_IP": "212.88.230.96", "HTTP_USER_AGENT": "Mozilla/5.0 (X11; Linux x86_6
4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.110 Safari/537.36", "HTTP_CONNECTION": "close", "SERVER_NAME": "51.254.197.51", "REMOTE_PORT": "5
8818", "wsgi.url_scheme": "https", "SERVER_PORT": "443", "HTTP_X_SCHEME": "https", "HTTP_X_REQUESTED_WITH": "XMLHttpRequest", "wsgi.input": "<gunicorn.http.
body.Body object at 0x7fe50693c650>", "HTTP_DNT": "1", "HTTP_HOST": "51.254.197.51", "wsgi.multithread": "False", "HTTP_ACCEPT": "application/json; version=
*", "wsgi.version": "(1, 0)", "RAW_URI": "/backends/0a0609af-7ca6-4254-8677-66d4b1b0ef56/?contents=_relations%2Cregular_domains&timestamp=1467190344677", "w
sgi.run_once": "False", "wsgi.errors": "<open file '<stderr>', mode 'w' at 0x7fe51d47b1e0>", "REMOTE_ADDR": "127.0.0.1", "HTTP_ACCEPT_LANGUAGE": "nl,en-US;q
=0.8,en;q=0.6", "gunicorn.socket": "<socket fileno=25 sock=127.0.0.1:8002 peer=127.0.0.1:58818>", "CONTENT_TYPE": "application/json", "wsgi.file_wrapper": "
gunicorn.http.wsgi.FileWrapper", "CSRF_COOKIE": "3a2k3xIdhG6iCCo9azX3K5Vo3tTj0zeH", "HTTP_ACCEPT_ENCODING": "gzip, deflate, sdch"}, "request": {"timestamp":
 "1467190344677", "contents": "_relations,regular_domains"}}
2016-06-29 10:52:25 06300 +0200 - ctrl-roub-01 - 27679/140621951014224 - api/middleware - 2030 - ERROR - An unhandled exception occurred:  
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 78, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 399, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 396, in dispatch
    response = handler(request, *args, **kwargs)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 427, in new_function
    return_value = f(*args, **kwargs)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 73, in new_function
    return f(*args, **kw)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 290, in new_function
    return Response(FullSerializer(object_type, contents=contents, instance=obj).data, status=status.HTTP_200_OK)
  File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 573, in data
    self._data = self.to_native(obj)
  File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 349, in to_native
    value = field.field_to_native(obj, field_name)
  File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 198, in field_to_native
    value = get_component(value, component)
  File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 56, in get_component
    val = getattr(obj, attr_name)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 339, in <lambda>
    fget = lambda s: s._get_dynamic_property(dynamic)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 408, in _get_dynamic_property
    return self._backend_property(data_loader, dynamic)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 898, in _backend_property
    dynamic_data = function()
  File "/opt/OpenvStorage/ovs/dal/hybrids/storagerouter.py", line 82, in _vmachines_guids
    for vdisk in VDiskList.get_in_volume_ids(storagedriver_client.list_volumes(str(storagedriver.storagedriver_id))):
ClusterNotReachableException:  
```

**Comment by khenderick** [comment]
Duplicates internal ticket OVS-4557. This is because the framework can't request information about volumes that are on a storagedriver that is offline.

**Comment by khenderick** [comment]
Solution: Implement new calls with ObjectRegistryClient
Implementation of openvstorage/volumedriver#56
This works around openvstorage/volumedriver#55's behavior

**Comment by khenderick** [comment]
Fixed by #740

**Comment by deepshirsath** [comment]
**PASSED** on **'Open vStorage: 2.7.1-fargo.3-1'** 

Open vStorage: 2.7.1-fargo.3-1
Open vStorage - Backend: 1.7.1-fargo.1-1
Open vStorage - SDM: 1.6.1-fargo.1-1
Storage Driver - Base: 6.0.3-dev.201607191821.0758481
Storage Driver - Server: 6.0.3-dev.201607191821.0758481
Alba: 0.9.14
Arakoon 1.9.8
- Storage router info on OVS gui was showing without any issues even though 'ovs-volumedriver_pool1' was HALTED.
- Also in 3 node cluster, 2nd node is HALTED/DOWN then also backend is loadable.

![backend-loding](https://cloud.githubusercontent.com/assets/14201912/17643497/8173cbd0-6189-11e6-94b7-724ddeafb7cc.PNG)

![node-2-down](https://cloud.githubusercontent.com/assets/14201912/17643504/ccfacc52-6189-11e6-88a7-5af53726885f.PNG)

![disk-status](https://cloud.githubusercontent.com/assets/14201912/17643505/d47d9194-6189-11e6-85b7-c7adae59de85.PNG)
